File: tests/arnaud_legoux_moving_average_test.py
import numpy as np

from taflow import ArnaudLegouxMovingAverage


def test_lifecycle() -> None:
    values = np.linspace(1.0, 20.0, 32)
    state = ArnaudLegouxMovingAverage(9, 0.85, 6.0).extend(values)
    assert len(state) == len(values)
    assert state.reset().value is None

File: crates/taflow-core/src/stream/arnaud_legoux_moving_average_test.rs
use super::arnaud_legoux_moving_average::ArnaudLegouxMovingAverage;

#[test]
fn warmup_and_reset_are_consistent() {
    let mut state = ArnaudLegouxMovingAverage::new(5, 0.85, 6.0).unwrap();
    for value in 0..8 {
        state.append(value as f64);
    }
    assert!(state.value().is_some());
    state.reset();
    assert_eq!(state.value(), None);
}
